### PR TITLE
Fix: custom key failover test flakiness

### PR DESCRIPTION
**Pull request by devcrafter**
`slowdown_count` is used only by hedge connections. But the counter is stored inside connection pools and was used for pools shuffling always. So, query execution which used hedged connections could affect connection load balancing for queries w/o hedged connections by updating `slowdown_count`

CC @Avogar 

Related to https://github.com/ClickHouse/ClickHouse/pull/58993#issuecomment-1900394520

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit bb5a6dd8d3ce4a8560d4ee1a515c4fb64492535c with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60158/bb5a6dd8d3ce4a8560d4ee1a515c4fb64492535c/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/60158/bb5a6dd8d3ce4a8560d4ee1a515c4fb64492535c/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Comment by Algunenano** [comment]
The flaky check is still not happy: https://s3.amazonaws.com/clickhouse-test-reports/60158/bb5a6dd8d3ce4a8560d4ee1a515c4fb64492535c/integration_tests_flaky_check__asan_/integration_run_flaky_0.log

Do you mind merging master please? Most of the failures (debug, tsan, upgrade) should be gone now.

**Comment by devcrafter** [comment]
> The flaky check is still not happy: https://s3.amazonaws.com/clickhouse-test-reports/60158/bb5a6dd8d3ce4a8560d4ee1a515c4fb64492535c/integration_tests_flaky_check__asan_/integration_run_flaky_0.log

Yes, see fix here - [a8e4627](https://github.com/ClickHouse/ClickHouse/pull/60158/commits/a8e4627663599a3fa722fdb470e058c18c79c933)

File: tests/integration/test_parallel_replicas_custom_key_failover/test.py [hunk 1]
```diff
@@ -108,12 +108,9 @@ def test_parallel_replicas_custom_key_failover(
...
         )
 
-        # currently this assert is flaky with asan and tsan builds, disable the assert in such cases for now
-        # will be investigated separately
-        if (
-            not node1.is_built_with_thread_sanitizer()
-            and not node1.is_built_with_address_sanitizer()
-        ):
+        # With enabled hedged requests, we can't guarantee exact query distribution among nodes
+        # In case of a replica being slow in terms of responsiveness, hedged connection can change initial replicas choice
```
Comment (Algunenano) [inline tests/integration/test_parallel_replicas_custom_key_failover/test.py:112]: :heart:

**Review by Algunenano — APPROVED** [review]

**Pull request merged**